Mathlib/Algebra/BigOperators/Fin.lean

```diff
@@ -388,6 +388,17 @@ theorem finPiFinEquiv_single {m : ℕ} {n : Fin m → ℕ} [∀ i, NeZero (n i)]
   rintro x hx
   rw [Pi.single_eq_of_ne hx, Fin.val_zero', zero_mul]
 
+/-- Equivalence between the Sigma type `(i : Fin m) × Fin (n i)` and `Fin (∑ i : Fin m, n i)`. -/
+def finSigmaFinEquiv {m : ℕ} {n : Fin m → ℕ} : (i : Fin m) × Fin (n i) ≃ Fin (∑ i : Fin m, n i) :=
```

Comment:
Could you please provide a few API lemmas that will help working with this defn?

Reply:
I added `finSigmaFinEquiv_apply` and `finSigmaFinEquiv_one`. I would add `finSigmaFinEquiv_symm_apply` but I don't think there's a particularly clean statement of the inverse equivalence (at least, not clean enough to be useful - I think the forward apply should be preferred)

Reply:
Note that there's WIP in Batteries to provide a more cohesive framework for `Fin` equivalences: https://github.com/leanprover-community/batteries/pull/1007.

My PR also defines auxiliary functions that explicitly identify the inverse equivalence #19013.

Reply:
I see, I guess one can write it in terms of these `divSum` and `modSum` you defined. Well, I guess we both have definitions of this equivalence. I'll leave mine like this (without a the `_symm_apply` lemma), I'm fine with either definition getting merged. #19695 just requires _some_ definition of finSigmaFinEquiv.

Reply:
I'm also fine with either definitions, and wonder if we could have the best of both worlds (explicit formulas from my PR, plus theorems that relate it to your PR).